api/envoy/extensions/filters/http/file_system_buffer/v3/file_system_buffer.proto

```diff
@@ -0,0 +1,131 @@
+  // ``default_memory_buffer_bytes_limit`` == 1024*1024
+  google.protobuf.UInt64Value memory_buffer_bytes_limit = 2;
+
+  // The maximum storage (excluding memory) to be buffered in this filter.
+  // If unset in route, vhost and listener config, defaults to a hardcoded value
+  // ``default_storage_buffer_bytes_limit`` == 32*1024*1024
+  google.protobuf.UInt64Value storage_buffer_bytes_limit = 3;
+
+  // The maximum amount that can be queued for writing to storage, above which the
+  // source is requested to pause. If unset, defaults to the same value as
+  // ``memory_buffer_bytes_limit``.
+  //
+  // For example, assuming the recipient is not consuming data at all, if
+  // ``memory_buffer_bytes_limit`` was 32*1024, and ``storage_buffer_queue_high_watermark_bytes``
+  // was 64*1024, and the filesystem is backed up so writes are not occurring promptly,
+  // then:
+  // * Any request less than 32*1024 will eventually pass through without ever attempting
+  //   to write to disk.
+  // * Any request with over 32*1024 buffered will start trying to write to disk.
+  //   * If it reaches (32+64)*1024 buffered and not yet written to disk, a high
+  //     watermark signal is sent to the source.
+  // * Any stream whose total size exceeds
+  //   ``memory_buffer_bytes_limit + storage_buffer_bytes_limit`` will provoke an error.
+  //   (Note, if the recipient *is* consuming data then it is possible for such an
+  //   oversized request to pass through the buffer filter, provided the recipient
+  //   isn't consuming data too slowly.)
+  //
+  // The low watermark signal is sent when the memory buffer is at size
+  //  ``memory_buffer_bytes_limit + (storage_buffer_bytes_limit / 2)``.
+  google.protobuf.UInt64Value storage_buffer_queue_high_watermark_bytes = 4;
+}
+
+// [#not-implemented-hide:]
+// Configuration for file system buffer filter.
+//
+// Route-specific configs override only the fields they explicitly include; unset
+// fields inherit from the vhost or listener-level config, or, if never set,
+// and not required, use a default value.
+message FileSystemBufferFilterConfig {
+  // A configuration for an AsyncFileManager.
+  //
+  // If unspecified in route, vhost and listener, an exception will be thrown.
+  common.async_files.v3.AsyncFileManagerConfig manager_config = 1;
+
+  // An optional path to which the unlinked files should be written - this may
+  // determine which physical storage device will be used.
+  //
+  // If unspecified in route, vhost and listener, will use the environment variable
+  // ``TMPDIR``, or, if that's unset, will use ``/tmp``.
+  google.protobuf.StringValue storage_buffer_path = 2;
+
+  // Optional configuration for how to buffer (or not) requests.
+  // See ``StreamConfig`` definition for defaults.
+  StreamConfig request = 3;
+
+  // Optional configuration for how to buffer (or not) responses.
+  // See ``StreamConfig`` definition for defaults.
+  StreamConfig response = 4;
```

Comment:
I still don't understand why we need "bypass" above. If either of these are not present, doesn't it mean bypass?

Reply:
A problem with doing it that way is that if at listener-level you set it to stream_if_possible, and then you want to set one route to bypass, how would you encode that in the config - unset for that route would mean "inherit the behavior from listener level". (In addition to making the default filter behavior "do nothing" which seems to not be common.)

Had another go at making the default behavior clearer in the comments.

Reply:
Yup that makes sense, thanks.